Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -249,22 +250,23 @@ - (void)saveUserProfile:(UserProfile *)userProfile
                                                     messageSender:self.messageSender
                                                    profileManager:self];
 
-                [[NSNotificationCenter defaultCenter] postNotificationName:kNSNotificationName_LocalProfileDidChange
-                                                                    object:nil
-                                                                  userInfo:nil];
+                [[NSNotificationCenter defaultCenter]
+                    postNotificationNameAsync:kNSNotificationName_LocalProfileDidChange
+                                       object:nil
+                                     userInfo:nil];
             } else {
                 [[NSNotificationCenter defaultCenter]
-                    postNotificationName:kNSNotificationName_OtherUsersProfileWillChange
-                                  object:nil
-                                userInfo:@{
-                                    kNSNotificationKey_ProfileRecipientId : userProfile.recipientId,
-                                }];
+                    postNotificationNameAsync:kNSNotificationName_OtherUsersProfileWillChange
```

Comment:
`OWSContactsManager` uses this to evacuate the avatar cache.  This can safely be done async, so long as it is posted before `kNSNotificationName_OtherUsersProfileDidChange`.